includes/functions.php

```diff
@@ -2371,4 +2056,645 @@ function tsml_date_localised($format, $timestamp = null)
+            } else {
+                $meeting_name_linked = $meeting_slug;
+            }
+            if (isset($meeting_time)) {
+                $meeting_time_formatted = date(get_option('time_format'), strtotime($meeting_time));
+            }
+
+            if ($is_matched) {
+                $message = __("<tr style='color:gray;'><td>Change</td><td >$meeting_name_linked</td><td>$day_of_week</td><td>$meeting_time_formatted</td><td>$last_update</td></tr>", '12-step-meeting-list');
+
+                //add changed record id to list of db records to be removed
+                $db_ids_to_delete[] = $db_meeting_id;
+            } else {
+                $message = __("<tr style='color:green;'><td>Add New</td><td >$meeting_name_linked</td><td>$day_of_week</td><td>$meeting_time_formatted</td><td>$last_update</td></tr>", '12-step-meeting-list');
+            }
+            $message_lines[] = $message;
+        }
+    }
+
+    //add to removal array (i.e. $db_ids_to_delete) with only those meetings in local database which are not matched with the import feed
+    foreach ($db_meetings as $db_meeting) {
+        list($day_of_week, $dow_number) = tsml_get_day_of_week_info($db_meeting['day'], $tsml_days);
+        $meeting_slug = $db_meeting['slug'];
+        $is_matched = in_array($meeting_slug, $feed_slugs);
+
+        // Check if slug has been modified on import by removing an appended suffix and test for match again
+        if (!$is_matched) {
+            for ($x = 0; $x <= 50; $x++) {
+                if (str_contains($meeting_slug, '-' . $x)) {
+                    $meeting_slug = str_replace('-' . $x, '', $meeting_slug);
+                    break;
+                }
+            }
+            $is_matched = in_array($meeting_slug, $feed_slugs);
+        }
+
+        if (!$is_matched) {
+            $current_update_time = date(get_option('date_format') . ' ' . get_option('time_format'), current_time('timestamp'));
+            $meeting_name = $db_meeting['name'];
+            $permalink = get_permalink($db_meeting['id']);
+            $meeting_name_linked = '<a href=' . $permalink . '>' . $meeting_name . '</a>';
+            $meeting_time = $db_meeting['time'];
+            $meeting_time_formatted = date(get_option('time_format'), strtotime($meeting_time));
+            $message = __("<tr style='color:red;'><td>Remove</td><td >$meeting_name_linked</td><td>$day_of_week</td><td>$meeting_time_formatted</td><td>$current_update_time</td></tr>", '12-step-meeting-list');
+            $message_lines[] = $message;
+            $db_ids_to_delete[] = $db_meeting['id'];
+        }
+    }
+
+    return $import_updates;
+}
+
+//function:	return boolean indicating whether or not the matched database and import records are in sync
+//used:		here (called from tsml_get_import_changes_only)
+function tsml_do_differential_comparison($db_meeting, $import_meeting, $data_source_last_update, $data_source_url)
```

Comment:
how about `tsml_compare_meetings`? and i don't think we need the last two arguments (see my comment below on data_source_url)

Reply:
$data_source_last_update certainly isn't needed anymore as we no longer do an incremental comparison. We can discuss the reasoning behind the need for $data_source_url.